seckill-command-service/src/main/java/io/servicecomb/poc/demo/seckill/web/SecKillCommandRestController.java

```diff
@@ -37,23 +37,29 @@
 
   private static final Logger logger = LoggerFactory.getLogger(SecKillCommandRestController.class);
 
-  private final List<SecKillCommandService<String>> commandServices;
+  private final Map<String, SecKillCommandService<String>> commandServices;
 
   @Autowired
-  public SecKillCommandRestController(List<SecKillCommandService<String>> commandServices) {
+  public SecKillCommandRestController(Map<String, SecKillCommandService<String>> commandServices) {
     this.commandServices = commandServices;
   }
 
   @RequestMapping(method = RequestMethod.POST, value = "/")
   public ResponseEntity<String> seckill(
       @RequestBody CouponDto<String> couponDto) {
     if (isValidCoupon(couponDto)) {
-      boolean result = commandServices.get(0).addCouponTo(couponDto.getCustomerId());
-      logger.info("SecKill from = {}, result = {}", couponDto.getCustomerId(), result);
-      if (result) {
-        return new ResponseEntity<>("Request accepted", HttpStatus.OK);
+      if (commandServices.containsKey(couponDto.getPromotionId())) {
+        int result = commandServices.get(couponDto.getPromotionId()).addCouponTo(couponDto.getCustomerId());
+        logger.info("SecKill from = {}, result = {}", couponDto.getCustomerId(), result);
+        if (result == 0) {
+          return new ResponseEntity<>("Request accepted", HttpStatus.OK);
+        } else if (result == 1) {
+          return new ResponseEntity<>("Request rejected due to coupon out of stock", HttpStatus.OK);
```

Comment:
we may want to use different http status code to distinguish between successful and rejected requests